### ISSUE TITLE
Test Failures with Ruby 3.0.0

### ISSUE THREAD

**Issue by JoshSinyor** [JoshSinyor/simplecov-shield#3]
When the environment is switched to Ruby 3.0.0, 7 of the RSpec tests fail. The first failed test is replicated below.

```
1) SimpleCov::Formatter::ShieldFormatter#generate_shield should call shields.io api
     Failure/Error: URI.encode(url)

     NoMethodError:
       undefined method `encode' for URI:Module
     # ./lib/simplecov-shield.rb:31:in `shield_url'
     # ./lib/simplecov-shield.rb:23:in `block in generate_shield'
     # ./lib/simplecov-shield.rb:22:in `open'
     # ./lib/simplecov-shield.rb:22:in `generate_shield'
     # ./spec/simplecov-shield_spec.rb:48:in `block (3 levels) in <top (required)>'
```

**Comment by JoshSinyor** [comment]
The Ruby module [`URI::Escape`](https://ruby-doc.org/stdlib-2.4.1/libdoc/uri/rdoc/URI/Escape.html) (alias `URI::Encode`, `URI::Decode`) is used to escape unsafe characters from code. This method [was deprecated](https://github.com/ruby/ruby/commit/238b979f1789f95262a267d8df6239806f2859cc) in Ruby 1.9.2 and is not included with Ruby 3.0.0.

The `URI::Escape` module's alias `URI::Encode` is used on line 31 of simplecov-shield.rb.

**Comment by JoshSinyor** [comment]
The `URI::Escape` module does not have a like-for-like replacement. Like-for-like switch for `CGI::Escape` results in full encoding of URLs, including forward slash and semi-colon characters.

Issue patched with switch to `CGI::EscapeElement` module, replacing only percentage character. Lines switched as follows:

`URI.encode(url)`

`CGI.escapeElement(url, "%")`

RSpec tests now pass (commit [7d94fff](https://github.com/JoshSinyor/simplecov-shield/commit/7d94fff6941ffdf7583580ee10725838777d3360)).
